lib/streamlit/testing/v1/element_tree.py

```diff
@@ -1556,7 +1558,53 @@ def run(self, *, timeout: float | None = None) -> AppTest:
         return self.root.run(timeout=timeout)
 
     def __repr__(self):
-        return util.repr_(self)
+        return repr_(self)
+
+
+def repr_(self) -> str:
+    classname = self.__class__.__name__
+
+    defaults: list[Any] = [None, "", False, [], set(), dict()]
+
+    if dataclasses.is_dataclass(self):
+        fields_vals = (
+            (f.name, getattr(self, f.name))
+            for f in dataclasses.fields(self)
+            if f.repr
+            and getattr(self, f.name) != f.default
+            and getattr(self, f.name) not in defaults
+        )
+    else:
+        fields_vals = ((f, v) for (f, v) in self.__dict__.items() if v not in defaults)
```

Comment:
nit: might be worth adding a comment above `defaults` mentioning that empty lists/sets/dicts compare to be equal -- I thought this wouldn't work when first reading through this before double checking that they do